addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -267,6 +267,9 @@ function factory(dependencies) {
          * @returns {mail.message[]}
          */
         async _loadMessages({ extraDomain, limit = 30 } = {}) {
+            if (this.isLoading) {
+                return;
+            }
```

Comment:
This does not feel good, it's already prevented in `loadMoreMessages` and `loadNewMessages`. And the compute of `hasToLoadMessages` should be taking care of setting itself to `true` only when it's appropriate to actually load. If there is a fix, it should be in that compute I think, or in one of the fields the compute is using, maybe `isCacheRefreshRequested`.